[dynamo] Add ddp_graphs artifact

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100021
* #100093

I want to be able to decouple DDP graph printing from the rest of
dynamo DEBUG-level logging, since frequently these logs are particularly
enlightening.

Differential Revision: [D45290919](https://our.internmc.facebook.com/intern/diff/D45290919/)

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

bertmaher committed:
[dynamo] Add ddp_graphs artifact

I want to be able to decouple DDP graph printing from the rest of
dynamo DEBUG-level logging, since frequently these logs are particularly
enlightening.

Differential Revision: [D45290919](https://our.internmc.facebook.com/intern/diff/D45290919/)

[ghstack-poisoned]

wconstab reviewed (approved):
i'm willing to stamp to unblock either way, but i'm wondering if you want to make the new topic more than just the single 'debug_str' - there are a few other logs inside DDPOptimizer that are still going to log.info, should they be going to ddp_log.info instead?  Did you leave it unchanged on purpose bc it works well in practice this way (regular info logs are on by default for you or something?)

mlazos commented on torch/_logging/_internal.py:
```diff
@@ -126,6 +126,7 @@ def set_logs(
     bytecode=None,
     aot_graphs=None,
     aot_joint_graph=None,
+    ddp_graphs=None,
```
An oss contributor added nice comments for each arg in a recent PR can you update those too to explain to users what these are?

mlazos commented:
Also there are a bunch of tests in dynamo/test_logging.py, can you add a test there?

bertmaher commented:
@wconstab Actually this is a good catch, there are a few other graphs that I didn't realize I had missed: children graphs of submodules, and the "final" graph.  In fact, what I think I'm really trying to achieve is just to be able to set the log level of distributed higher than that of the rest of dynamo.  (Primarily because the debug-level bytecode dump/trace is too big, which is what the previous PR was tackling)

bertmaher committed:
Update on "[dynamo] Add ddp_graphs artifact"


I want to be able to decouple DDP graph printing from the rest of
dynamo DEBUG-level logging, since frequently these logs are particularly
enlightening.

Differential Revision: [D45290919](https://our.internmc.facebook.com/intern/diff/D45290919/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

bertmaher committed:
Update on "[dynamo] Add ddp_graphs artifact"


I want to be able to decouple DDP graph printing from the rest of
dynamo DEBUG-level logging, since frequently these logs are particularly
enlightening.

Differential Revision: [D45290919](https://our.internmc.facebook.com/intern/diff/D45290919/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

bertmaher commented:
@pytorchbot merge

bertmaher committed:
filter test records on "[dynamo] Add ddp_graphs artifact"


I want to be able to decouple DDP graph printing from the rest of
dynamo DEBUG-level logging, since frequently these logs are particularly
enlightening.

Differential Revision: [D45290919](https://our.internmc.facebook.com/intern/diff/D45290919/)

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

bertmaher commented:
@pytorchbot merge
